### PR TITLE
3016 lesmis chart loading states

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Remove loading indicator from year selector
- Add background fetching indicator

---

### Screenshots:
see issue

### CODE REVIEW

File: packages/lesmis/src/components/Chart.js [hunk 1]
```diff
@@ -72,21 +74,24 @@ const Toggle = ({ value, onChange }) => (
...
-      </ChartWrapper>
-    ) : (
-      <Wrapper>
-        <Table viewContent={viewContent} />
-      </Wrapper>
-    )}
-  </FetchLoader>
-);
+const ChartTable = ({ viewContent, isLoading, isFetching, isError, error, selectedTab }) => {
+  const showIsLoading = isLoading || (!getIsChartData(viewContent) && isFetching);
```
Comment (tcaiger) [inline packages/lesmis/src/components/Chart.js:78]: Also show the full loader if there is no data and we are fetching new data in the backgrond

Comment (edmofro) [reply packages/lesmis/src/components/Chart.js:78]: Nice touch, so we keep the animations is there is already data but show a more obvious loader when there is no data 👌

File: packages/lesmis/src/components/Chart.js [hunk 2]
```diff
@@ -111,13 +116,19 @@ export const Chart = ({ name, viewContent, isLoading, isError, error, isEnlarged
   ) : (
     <>
       <Header>
-        <Title>{name}</Title>
+        <FlexStart>
+          <Title>{name}</Title>
+          {isFetching && !isLoading && getIsChartData(viewContent) && (
```
Comment (edmofro) [inline packages/lesmis/src/components/Chart.js:121]: Seems the same slightly complicated decision is being made in two places, so could be better to pull out a variable `isFullLoading`* within this component and then just pass that value through as a prop to the `ChartTable` as `isLoading`, and here use
```suggestion
          {isFetching && !isFullLoading && (
```
*could do with a better name

**Review by edmofro — CHANGES_REQUESTED** [review]
One request, happy for push back. Visual placement looks great :-)

File: packages/lesmis/src/components/Chart.js (hunk 2 above)
Comment (tcaiger) [reply packages/lesmis/src/components/Chart.js:121]: I agree it does seem like they could be combined but I'm not quite sure how to combine the two decisions as the logic for them is actually different. Maybe I'm not seeing a way of doing it?
eg.
`isFetching && !isLoading && getIsChartData(viewContent)` vs `isLoading || (!getIsChartData(viewContent) && isFetching)`

Comment (edmofro) [reply packages/lesmis/src/components/Chart.js:121]: Would this work? It doesn't combine them all into one variable, which I also can't see a way to do, but pulls the logic to all be in one place https://github.com/beyondessential/tupaia/pull/2779 (and reuses the tricky decision of "fetching but from no data state, so mark as a full load")

**Review by edmofro — APPROVED** [review]

**Pull request merged**